src/Zicht/Bundle/AdminBundle/Command/ExportSqlCommand.php

```diff
@@ -99,21 +102,230 @@ protected function execute(InputInterface $input, OutputInterface $output)
             $file = 'php://output';
         }
 
-        Handler::create(
-            new DoctrineDBALConnectionSourceIterator(
-                $this->getConnection(),
-                $input->getArgument('sql')
-            ),
-            new $this->typeMapping[$input->getOption('type')]($file)
-        )
-            ->export();
+        $this->getHandler($file, $input)->export();
+    }
+
+    /**
+     * @param InputInterface $input
+     * @return Handler
+     */
+    private function getHandler($file, InputInterface $input)
+    {
+        return Handler::create(
+            $this->getSourceIteratorInter($input->getArgument('sql')),
+            $this->getWriter($input->getOption('type'), $file, $input)
+        );
+    }
+
+
+    /**
+     * @param $type
+     * @param $file
+     * @param InputInterface $input
+     * @return WriterInterface
+     */
+    private function getWriter($type, $file, InputInterface $input)
+    {
+        return new $this->typeMapping[$type]($file, ...$this->getArgs($type, $input));
+    }
+
+    /**
+     * @return SourceIteratorInterface
+     */
+    private function getSourceIteratorInter($sql)
+    {
+        return new DoctrineDBALConnectionSourceIterator($this->getConnection(), $sql);
     }
 
     /**
      * @return Connection
      */
-    protected function getConnection()
+    private function getConnection()
```

Comment:
dit is een breaking change :)

Reply:
It is a code change in a Command file... strictly speaking you are right, but in this case I would prefer to consistently use private.